backend/fms_core/migrations/0065_v4_12_0.py

```diff
@@ -28,6 +28,73 @@ def set_measured_volume_properties_optional(apps, schema_editor):
+        },
+        {"name": "Sample QC (RNA)", "protocol_name": "Sample Quality Control","expected_sample_type": "EXTRACTED_SAMPLE",
+          "specifications": [
+            {"display_name": "SampleQcType", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "RNA"}]
+        }
+    ]
+    StepOrder = apps.get_model("fms_core", "StepOrder")
+    SampleNextStep = apps.get_model("fms_core", "SampleNextStep")
+    Step = apps.get_model("fms_core", "Step")
+    Protocol = apps.get_model("fms_core", "Protocol")
+    StepSpecification = apps.get_model("fms_core", "StepSpecification")
+    # create the new steps and its necessary objects
+    with reversion.create_revision(manage_manually=True):
+        admin_user = User.objects.get(username=ADMIN_USERNAME)
+        reversion.set_comment(f"Create Sample QC DNA and RNA step and transfer existing Sample QC data to their respective new step.")
+        reversion.set_user(admin_user)
+        for step_info in STEPS:
+            protocol = Protocol.objects.get(name=step_info["protocol_name"])
+            step = Step.objects.create(name=step_info["name"],
+                                       protocol_id=protocol.id,
+                                       type="PROTOCOL",
+                                       needs_placement=False,
+                                       needs_planning=False,
+                                       expected_sample_type=step_info["expected_sample_type"],
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        oldStep = Step.objects.get(name="Sample QC")
+        dnaStep = Step.objects.get(name="Sample QC (DNA)")
+        rnaStep = Step.objects.get(name="Sample QC (RNA)")
+        # Start for data transfer process
+        sns = SampleNextStep.objects.filter(step__id=oldStep.id)
+        for sampleNextStep in sns:
+            updatedSampleNextStep = SampleNextStep.objects.get(id=sampleNextStep.id)
+            if sampleNextStep.sample.derived_samples.first().sample_kind.name == "DNA":
```

Comment:
Optional: I'm personally biased towards obtaining DNA `SampleKind` directly and doing something like
```python
sampleNextStep.sample.derived_samples.first().sample_kind.id == dna_sample_kind.id
```
It feels more robust compared to string-based comparison.